docs/src/references/synchronize-daemons.md

```diff
@@ -0,0 +1,116 @@
+- Now the mapper picks up this message and creates the c8y_LogfileRequest operation file in `/etc/tedge/operations/c8y`.
+- Also the mapper creates the `c8y-log-plugin.toml` file, which contains the `supported log types` in `/etc/tedge/c8y`.
+- Mapper sends the updated supported operations list to the c8y cloud. Also, it sends the supported `log types`(118,software-management) to the c8y cloud.
+
+Pros: 
+-   This will remove the file system dependency when the daemons want to create the supported operation files on the mapper file system.
+    For example, tedge-agent wants to create the supported operations files in /etc/tedge/operations/c8y directory.
+
+Cons: 
+-   This leads to hard dependency on the list of daemons in the mapper. For example, the tedge-mapper-c8y has to subscribe to tedge/health/tedge-agent
+topic to receive the health status from the tedge-agent. Based on this the mapper has to create a supported operation file in /etc/tedge/operations/c8y.
+-   This will be a hard-coded dependency of the name of the plugin and can’t be updated on the fly if someone wants to change the name of the daemon
+    or wants to add a new daemon.
+
+## Proposal 2:  Introduce an `init message` mechanism
+
+Make all the tedge components publish a `retained init` message describing their role (software management, config management, etc)
+ and other details specific to that role.
+
+The proposed topic will be `tedge/<device-id>/init/<cloud>`.
+Here `<cloud>` could be c8y, azure, aws, etc
+
+The message template is {Operation_name: <content>, Operation_name:<content>, Config_types: type1..typen}.
+Here `Operation_name` is the name of the operation to be published to the cloud and 
+the `content` is the content of the operation file, if any.
+Here `Config_types` is the configuration types that are supported by that particular daemon.
+
+Example 1:
+- When the tedge-mapper-c8y is started, it will subscribe to `tedge/<device-id>/init/#` topic
+- When tedge-agent is started, it will publish `{}` an empty `init` message on to `tedge/init/c8y/tedge-agent`
+- Now the mapper picks up this message and creates the `c8y_Restart, c8y_SoftwareUpdat` supported operations file in `/etc/tedge/operations/c8y`. 
+- Mapper sends the updated supported operations list to the c8y cloud.
+- Also, now that the tedge-agent is up and running, the tedge-mapper-c8y will publish the request to get the
+ software list on `tedge/commands/req/software/list`.
+- Now the tedge-agent processes this request and sends the software list to the mapper, once the mapper gets the response it will forward it to the c8y cloud.
+
+Example 2,
+- When tedge-mapper-c8y is started, it will subscribe to `tedge/<device-id>/init/#` topic
+- When c8y-log-plugin is started, it will publish {"LogfileRequest":"", "LogTypes": "software-management, mosquitto-log"} on to `tedge/thin-edge/init` topic
```

Comment:
```suggestion
- When c8y-log-plugin is started, it will publish {"c8y_LogfileRequest":"", "c8y_SupportedLogs": "software-management, mosquitto-log"} on to `tedge/thin-edge/init` topic
```
If we're supporting cloud specific endpoints anyway, then I'd keep the keys as close to the cloud supported ones so that the developer doesn't need to learn an extra one.  Sending the operation type is still relevant as the mapper needs to combine that with other supported operations as well. But, we can even skip keys like `c8y_SupportedLogs` as there is no benefit in making the mapper send them. The plugin could send it directly.

Reply:
If `plugin` has to send `c8y_SupportedLogs` then the plugin must be aware of the cloud. But here, we want to make the `plugins` cloud agnostic.

Reply:
Ideally the `init` command should be as cloud agnostic as possible, so I would also be in favour of the payload not having cloud specific names inside it. We can still use the Cumulocity IoT contract as a good reference point since the other Clouds don't offer these features out of the box.

Reply:
Oh don't get me wrong. I also want all our plugins to be as cloud-agnostic as possible. I was saying that it'd be better to keep the keys similar to that of the cloud **only** for the cloud specific init topics. If the `c8y-log-plugin` is converted into a cloud agnostic `tedge-log-plugin`, we wouldn't need it.